lib/ecto/query/builder/windows.ex

```diff
@@ -160,7 +160,14 @@ defmodule Ecto.Query.Builder.Windows do
   end
 
   defp do_runtime_window!([{:order_by, order_by} | kw], query, acc, params) do
-    {order_by, params} = OrderBy.order_by_or_distinct!(:order_by, query, order_by, params)
+    {order_by, params, subqueries, aliases} = OrderBy.order_by_or_distinct!(:order_by, query, order_by, params)
+
+    if subqueries != [] || aliases != %{} do
+      # TODO?
+      raise ArgumentError,
+            "subqueries and aliases are not allowed in runtime windows"
```

Comment:
I think I still need to address this as well, because we ought to support everything we support in order_bys in windows.

Reply:
I don't mind doing this separately, but AshPostgres does actually rely on the fact that order_by & windows support the same syntax, so I'll need to follow up with supporting this in windows soon after.

Reply:
I see, as you prefer then!

Reply:
Always better to limit scope though, so I will do it in a separate PR when we're done here :)

Reply:
Ended up doing it in this PR after all :)